components/base/BaseTag.vue

```diff
@@ -43,289 +43,75 @@ const props = withDefaults(
-
-            break
-          }
-          case 'pastel': {
-            classes.push(
-              'bg-primary-100 text-primary-500 border-primary-100 dark:border-primary-500 dark:text-primary-500 border dark:bg-transparent'
-            )
-
-            break
-          }
-          case 'outline': {
-            classes.push('text-primary-500 border-primary-500 border ')
-
-            break
-          }
-        }
-        break
-      }
-      case 'info': {
-        switch (props.flavor) {
-          case 'solid': {
-            classes.push('bg-info-500 dark:bg-info-500 text-white')
-
-            break
-          }
-          case 'pastel': {
-            classes.push(
-              'bg-info-100 text-info-500 border-info-100 dark:border-info-500 dark:text-info-500 border dark:bg-transparent'
-            )
-
-            break
-          }
-          case 'outline': {
-            classes.push('text-info-500 border-info-500 border')
-
-            break
-          }
-        }
-        break
-      }
-      case 'success': {
-        switch (props.flavor) {
-          case 'solid': {
-            classes.push('bg-success-500 dark:bg-success-500  text-white')
-
-            break
-          }
-          case 'pastel': {
-            classes.push(
-              'bg-success-100 text-success-500 border-success-100 dark:border-success-500 dark:text-success-500 border dark:bg-transparent'
-            )
-
-            break
-          }
-          case 'outline': {
-            classes.push('text-success-500  border-success-500 border')
-
-            break
-          }
-        }
-        break
-      }
-      case 'warning': {
-        switch (props.flavor) {
-          case 'solid': {
-            classes.push('bg-warning-500 dark:bg-warning-500 text-white')
-
-            break
-          }
-          case 'pastel': {
-            classes.push(
-              'bg-warning-100  text-warning-500 border-warning-100 dark:border-warning-500 dark:text-warning-500 border dark:bg-transparent'
-            )
-
-            break
-          }
-          case 'outline': {
-            classes.push('text-warning-500  border-warning-500 border')
+const solidColorStyle = computed(() => [
+  `bg-${props.color}-500`,
+  `dark:bg-${props.color}-500`,
+  `text-white`,
+])
+const pastelColorStyle = computed(() => [
+  `bg-${props.color}-100`,
+  `text-${props.color}-500`,
+  `border-${props.color}-100`,
+  `dark:border-${props.color}-500`,
+  `dark:text-${props.color}-500`,
+  `border`,
+  `dark:bg-transparent`,
+])
+const outlineColorStyle = computed(() => [
+  `text-${props.color}-500`,
+  `border-${props.color}-500`,
+  `border`,
+])
+const colorStyle = computed(() => {
+  let colors = {}
+  let styleToUse =
+    props.flavor === 'solid'
+      ? solidColorStyle.value
+      : props.flavor === 'outline'
+      ? outlineColorStyle.value
+      : pastelColorStyle.value
+
+  Reflect.set(colors, props.color, styleToUse)
```

Comment:
What's the benefit of using Reflect API here? It's way much slower than using classic setter

<img width="558" alt="image" src="https://user-images.githubusercontent.com/3911343/235930955-19c79b6b-113a-4a7a-b9cf-649c609f02ba.png">